### PR TITLE
[4]send email on new user via webservice

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #37128 .

### Summary of Changes
fix send email when the user is created via webservices

### Testing Instructions
create a new user POST `{{base_path}}/api/index.php/v1/users`

```json
{
    "block": "0",
    "email": "apipr@mail.com",
    "groups": [
        "2"
    ],
    "id": "0",
    "lastResetTime": "",
    "lastvisitDate": "",
    "name": "apipr",
    "params": {
        "admin_language": "",
        "admin_style": "",
        "editor": "",
        "helpsite": "",
        "language": "",
        "timezone": ""
    },
    "password": "qwertyqwerty123",
    "password2": "qwertyqwerty123",
    "registerDate": "",
    "requireReset": "0",
    "resetCount": "0",
    "sendEmail": "1",
    "username": "testAPIpr"
}
```

### Actual result BEFORE applying this Pull Request
no email was sent

### Expected result AFTER applying this Pull Request
email sent to the created new user

### CODE REVIEW

**Comment by nbradshaw45** [comment]
Thank you! This tested good!

However, I have a suggestion:

"sendEmail": (0 or 1) relates to receiving admin messages (the name is a little confusing).

I think it would make sense to add a new flag for send email to users on create - so that it is optional - there could be reasons why an admin may not want to send these emails.

So creating a new json element called "sendUserWelcomeEmail": values (0 or 1)

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 76237c09301196912842541cd31bcd52aa4b7dff<br /><br />Tested successfully with Postman in 4.1.1-dev of 27 February in Wampserver 3.2.7 using PHP 8.0.15.  The email was sent, received of course a non-delivery message to the Super User mailbox.  The response:
{"links":{"self":"http:\/\/joomla4test\/api\/index.php\/v1\/users"},"data":{"type":"users","id":"809","attributes":{"groups":{"2":2},"id":809,"name":"apipr","username":"testAPIpr","email":"apipr@mail.com","block":0,"sendEmail":1,"registerDate":"2022-02-27 10:12:07","lastvisitDate":null,"lastResetTime":null,"resetCount":0}}}<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

File: plugins/user/joomla/joomla.php [hunk 1]
```diff
@@ -152,7 +152,7 @@ public function onUserAfterSave($user, $isnew, $success, $msg): void
 
 		// @todo: Suck in the frontend registration emails here as well. Job for a rainy day.
 		// The method check here ensures that if running as a CLI Application we don't get any errors
-		if (method_exists($this->app, 'isClient') && !$this->app->isClient('administrator'))
+		if (method_exists($this->app, 'isClient') && ($this->app->isClient('frontend') || $this->app->isClient('cli')))
```
Comment (wilsonge) [inline plugins/user/joomla/joomla.php]:
```suggestion
		if (method_exists($this->app, 'isClient') && ($this->app->isClient('site') || $this->app->isClient('cli')))
```

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 585690c01c805a81869b8d6b47c085156dcd959e<br /><br /># Tested successfully with Postman in 4.1.1-dev of 28 February in Wampserver 3.2.7 using PHP 8.0.15. The email was sent all right.  
The response:
{"links":{"self":"http:\/\/joomla4test\/api\/index.php\/v1\/users"},"data":{"type":"users","id":"811","attributes":{"groups":{"2":2},"id":811,"name":"apipr","username":"testAPIpr","email":"xxxxx@example.com","block":0,"sendEmail":1,"registerDate":"2022-02-28 08:58:38","lastvisitDate":null,"lastResetTime":null,"resetCount":0}}}<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

**Comment by Quy** [comment]
2nd successful test https://github.com/joomla/joomla-cms/issues/37128#issuecomment-1052607179

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on b09c1fba37ff31fc38e624d14b4439fca932ca58<br /><br />Tested successfully - for the third time in 4.1.1-dev of 8 March in Wampserver 3.2.7 using PHP 8.0.15
The email was sent auccessfully.  Response:
{"links":{"self":"http:\/\/joomla4test\/api\/index.php\/v1\/users"},"data":{"type":"users","id":"309","attributes":{"groups":{"2":2},"id":309,"name":"apipr","username":"toivo.user4","email":"admin@xxxxxx.com","block":0,"sendEmail":1,"registerDate":"2022-03-08 06:56:33","lastvisitDate":null,"lastResetTime":null,"resetCount":0}}}<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on b09c1fba37ff31fc38e624d14b4439fca932ca58<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37149">issues.joomla.org/tracker/joomla-cms/37149</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**